src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final MovieSessionService movieSessionService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> responseMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  MovieSessionService movieSessionService,
+                                  DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          responseMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.responseMapper = responseMapper;
+    }
+
+    @PutMapping("/movie-session")
+    public void addMovieSession(@RequestBody Long userId,
+                                @RequestBody Long movieSessionId) {
```

Comment:
![image](https://user-images.githubusercontent.com/104957547/180003574-6afa4f98-2d68-4fa7-b9a1-ad098dbbd957.png)
pay attention to this reqiuirement. It isn't correct implementation. We use @RequestBody annotation in case when we transfer DtoRequestObject using POST method. 

But in this case we should use @RequestParam annotation. It is intended for sending input params in url but not in the body of request. Example:

http://localhost:8080/movie-session?userId=1&movieSessionId=1

Reply:
Thank's for this. Now I understood=)